aquadoggo/src/db/stores/operation.rs

```diff
@@ -0,0 +1,551 @@
+                query(
+                    "
+                    INSERT INTO
+                        previous_operations_v1 (
+                            parent_operation_id,
+                            child_operation_id
+                        )
+                    VALUES
+                        ($1, $2)
+                    ",
+                )
+                .bind(prev_op_id.as_str())
+                .bind(operation.id().as_str().to_owned())
+                .execute(&self.pool)
+            }))
+            .await
+            // If any of database errors occur we will catch that here
+            .map_err(|e| OperationStorageError::Custom(e.to_string()))?;
+
+        // Same pattern as above but now for operation_fields. Construct and execute the
+        // queries, return their futures and execute all of them with `try_join_all()`.
+        let fields_insertion_result = match operation.fields() {
+            Some(fields) => {
+                let result = try_join_all(fields.iter().flat_map(|(name, value)| {
+                    // If the value is a relation_list or pinned_relation_list we need to insert a new field row for
+                    // every item in the list. Here we collect these items and return them in a vector. If this operation
+                    // value is anything except for the above list types, we will return a vec containing a single item.
+                    let db_values = parse_value_to_string_vec(value);
+
+                    // Collect all query futures.
+                    db_values
+                        .into_iter()
+                        .map(|db_value| {
+                            // Compose the query and return it's future.
+                            query(
+                                "
+                            INSERT INTO
+                                operation_fields_v1 (
+                                    operation_id,
+                                    name,
+                                    field_type,
+                                    value
+                                )
+                            VALUES
+                                ($1, $2, $3, $4)
+                            ",
+                            )
+                            .bind(operation.id().as_str().to_owned())
+                            .bind(name.to_owned())
+                            .bind(value.field_type().to_string())
+                            .bind(db_value)
+                            .execute(&self.pool)
+                        })
+                        .collect::<Vec<_>>()
+                }))
+                .await
+                // If any queries error, we catch that here.
+                .map_err(|e| OperationStorageError::Custom(e.to_string()))?;
+                Some(result)
+            }
+            None => None,
+        };
+
+        // Check every insertion performed affected exactly 1 row.
```

Comment:
I don't know when this situation occurs, and I don't know how to replicate it in tests :rofl: does anyone have any ideas?